typescript/nomad-deploy/src/incremental/index.ts

```diff
@@ -0,0 +1,63 @@
+import { NomadContext } from '@nomad-xyz/sdk/';
+import { canonizeId } from '@nomad-xyz/sdk/utils';
+
+/**
+ * Prepares and executes necessary calls to governing
+ * router for enrolling a spoke after core and
+ * bridge have been deployed
+ * @param sdk SDK containing new spoke domain
+ * @param spokeDomain domain of the spoke
+ * @param watchers set of watchers to be enrolled
+ */
+export async function enrollSpoke(
+  sdk: NomadContext,
+  spokeDomain: number,
+  watchers: string[],
+): Promise<void> {
+  let hubCore = await sdk.governorCore();
+  let hubBridge = sdk.mustGetBridge(hubCore.domain);
+
+  let spokeCore = await sdk.mustGetCore(spokeDomain);
+  let spokeBridge = await sdk.mustGetBridge(spokeDomain);
+  let batch = await hubCore.newGovernanceBatch();
+
+  // enroll watchers
+  await Promise.all(
+    watchers.map(async (watcher) => {
+      const call =
+        await hubCore.xAppConnectionManager.populateTransaction.watcherPermission(
```

Comment:
❗ this should be `setWatcherPermission` - just `watcherPermission` is a view call - how were the tests passing?

Reply:
ah. i see why the tests were passing. because we were passing in empty array of watchers, so we never checked that

Reply:
fixed: https://github.com/nomad-xyz/nomad-monorepo/pull/17/commits/5af04c7455f91b2ca142fe54adaa2c315f47ad28

 @kekonen need to re-run the local environment with these changes and ensure it passes - we are now properly enrolling watchers & testing that in the local environment